### ISSUE TITLE
adding downloadable tsv file of propagated gene annotations

### ISSUE THREAD

**Issue by ChristopherMancuso** [monarch-initiative/monarch-app#505]
The monarch web site currently offers tsv files for gene-phenotype associations for the major model species, which is an amazing resource! The files I reference are found at 
https://data.monarchinitiative.org/latest/tsv/gene_associations/index.html

However I don't believe these gene annotations are propagated (i.e. annotations to more specific terms are then also annotated to any ancestor node). I was wondering if you all at Monarch could provide a file of that nature? The idea of this is to be able to do species specific gene set enrichment on a collection of phenotypes.

Thanks!

**Comment by kevinschaper** [comment]
This is an interesting idea! 

I'll transfer this issue over for prioritization etc, but in the meantime, if you want to experiment with our sqlite artifact (https://data.monarchinitiative.org/monarch-kg/latest/monarch-kg.db.gz), you could try this query:

```
select edges.subject, edges.predicate, closure.object as object, edges.negated,
       case when edges.object = closure.object then true else false end as direct
from edges
    join closure on closure.subject = edges.object
where edges.predicate = 'biolink:has_phenotype'
```

or to add more detail about nodes on either side of the association, use the much wider denormalized_edges table: 

```
select edges.subject, edges.subject_label, subject_taxon, edges.predicate, closure.object as object, phenotype.name as object_label, edges.negated,
       case when edges.object = closure.object then true else false end as direct
from denormalized_edges as edges
    join closure on closure.subject = edges.object
    join nodes as phenotype on phenotype.id = closure.object
where edges.predicate = 'biolink:has_phenotype'
```

**Comment by ChristopherMancuso** [comment]
@glass-ships @kevinschaper I see this is closed now. is there a spot on the Monarch site I go to to download the propagated annotations?

**Comment by glass-ships** [comment]
I'm so sorry @ChristopherMancuso I thought I'd commented on this! 

We've implemented changes so that the association table for entities are in TSV format. 

Unfortunately I prematurely closed this issue, as those changes don't address the need of propagated gene annotations. My apologies! We'll continue work on this and update you when we have something that actually suits your use case

**Comment by ChristopherMancuso** [comment]
Thanks @glass-ships! No worries on anything, this propagated annotations problem seems quite hard! I really appreciate you putting time into this!

**Comment by kevinschaper** [comment]
Hi @ChristopherMancuso, we accidentally brought this issue in thinking it was one of several we have for adding tsv downloads from disease/gene/phenotype pages. I'll move this into a later milestone. 

I think our first experiment will be to see how large a propagated annotation file would be.

**Comment by ChristopherMancuso** [comment]
Sounds good!

**Comment by kevinschaper** [comment]
I hate to bump this again, but unfortunately I might need to. There's a bit of technical debt/pipeline re-organization going on, and additionally I want to be sure that it's not producing anything inaccurate or misleading. 

I'm working in https://github.com/monarch-initiative/monarch-ingest/pull/588 

Right now it looks like the scale is going to be ok, at least now that we have duckdb in the stack. I think exploding gene to phenotype produces about ~30m edges, and disease to phenotype is blowing up into about 15m.

**Comment by ChristopherMancuso** [comment]
Sounds good @kevinschaper! I was able to use the data that you pointed me to here https://data.monarchinitiative.org/monarch-kg/latest/tsv/gene_associations/index.html and then I downloaded all the .obo files for the species specific ontologies. Then for each species I was able to use the obo files to propagate the annotations. It is much easier as now the Monarch annotation are all for a single ontology per species. Before when I was using https://data.monarchinitiative.org/latest/tsv/gene_associations/index.html there was a mix of different ontology in each tsv file which was confusing how to handle

**Comment by ChristopherMancuso** [comment]
If this issue is open just because of me I would say you could close it

**Comment by kevinschaper** [comment]
I added the code to generate the propagated annotations, and it ran fine for me locally (under a minute!), but when I did a build on our cloud build infrastructure, it spent 12+ hours on this step before I killed it. I think at some point a better way to handle this would be with a notebook for how to download the duckdb version of the kg, create the propagated table and then write it to a file. For now, I ended up rewriting our export pipeline to pull from duckdb, but turned off the propagated exports to get a build to finish.
